src/transformers/models/prophetnet/modeling_prophetnet.py

```diff
@@ -2313,5 +2326,8 @@ def __init__(self, config: ProphetNetConfig):
         super().__init__(config)
         self.decoder = ProphetNetDecoder(config)
 
+        # This is a link so that tied weights work across classes
+        self.word_embeddings = self.decoder.word_embeddings
```

Comment:
As far as I know, this is necessary as `safetensors` doesn't want to save tied weights and therefore saves a single identifier for all the tied weights.

It becomes an issue when we have a few: in this case we have 4 tied weights, with some being loaded in some models (like encoders), and others being loaded in other models (like decoders).

If the encoder-decoder parent class saves a checkpoint, then it can select to save a single copy of a tensor which is only visible in the encoder; so when loading the decoder, it would discard that weight even though it is the only reference to the tied weights.